Pipelines/Templates/update-deployment-settings.yml

```diff
@@ -36,16 +37,18 @@ steps:
       $segments = ([System.Uri]"$(Build.Repository.Uri)").LocalPath.Split("/")
```

Comment:
@mikefactorial I suggest we use the `Segments` property on `Uri` instead of parsing `LocalPath`.

Then I suggest we use `Select-Object` to get the last three segments, and strip trailing `/` from them

```pwsh
[string[]]$segments = ([System.Uri]"$(Build.Repository.Uri)").Segments | Select-Object -Last 3 | ForEach-Object { $_.TrimEnd("/") }
```

In the if clause below we can now simply use `$segments[0]` and `$segments[2]` for Project and Repo regardless of uri structure.

i.e.

```pwsh
# Use the parsed value from the repo uri if it follows the expected format
if ($segments.Length -gt 2) {
  $pipelineProject = $segments[0]
  $pipelineRepo = $segments[2]
}
```

Reply:
Thanks for this @fredrikhr. As always appreciate the feedback. Will incorporate this into the PR